lib/mint/http.ex

```diff
@@ -377,6 +377,17 @@ defmodule Mint.HTTP do
   def upgrade(old_transport, transport_state, scheme, hostname, port, opts),
     do: Mint.Negotiate.upgrade(old_transport, transport_state, scheme, hostname, port, opts)
 
+  @doc """
+  Returns an atom describing the current connection type
+
+  ## Examples
+
+      :http1 = Mint.HTTP.protocol(conn)
+  """
+  @spec protocol(t()) :: :http1 | :http2
+  def protocol(%Mint.HTTP1{}), do: :http1
+  def protocol(%Mint.HTTP2{}), do: :http2
```

Comment:
I think we need to add a clause for `Mint.UnsafeProxy` conns:

```suggestion
  def protocol(%Mint.HTTP2{}), do: :http2
  def protocol(%Mint.UnsafeProxy{state: internal_conn}), do: protocol(internal_conn()
```

right @ericmj?